charmcraft/commands/store/store.py

```diff
@@ -184,10 +187,10 @@ def upload(self, name, filepath):
         endpoint = '/v1/charm/{}/revisions'.format(name)
         return self._upload(endpoint, filepath)
 
-    def upload_resource(self, charm_name, resource_name, filepath):
+    def upload_resource(self, charm_name, resource_name, resource_type, filepath):
         """Upload the content of filepath to the indicated resource."""
         endpoint = '/v1/charm/{}/resources/{}/revisions'.format(charm_name, resource_name)
-        return self._upload(endpoint, filepath)
+        return self._upload(endpoint, filepath, extra_fields={'type': resource_type})
```

Comment:
what happened to reading the value for metadata?

Reply:
Don't know, will find out,